platforms/riscv32/features/BUILD.bazel

```diff
@@ -54,6 +56,23 @@ feature(
     provides = ["compilation_mode"],
 )
 
+feature(
+    name = "guards",
+    enabled = True,
```

Comment:
This enables this compiler feature universally across the entire codebase.  Is that what is really intended?

As an alternative, I might suggest we add this to one of the compilation modes (or create a new compilation mode) such that hardening features can be turned on/off per bazel invocation.  Right now, the default build mode for development and testing is `fastbuild`.  To build software for release, we should be using build mode `opt` or perhaps even a custom build mode.

Reply:
Regarding the use of a compilation mode, I'm concerned that if we only added this to, say, the `opt` build mode that any potential problems caused by the hardening would not be detected until we did such a build. If we do want to use this, I suppose it should be part of the mode that we use for testing, including CI. I am OK with adding modes that do not include this, but I'm not sure if people would actually use those other modes...?

Regarding enabling this "across the entire codebase" or in a more localized fashion, you'll have to help me with the Bazel configuration (in the OpenTitan repo, I presume) as I don't know what's the correct way to make this more localized. Do you have any suggestions on what we should include and what we should exclude? The more I think about that the less clear the answer seems :)